std/algorithm/mutation.d

```diff
@@ -562,6 +562,17 @@ void fill(Range, Value)(Range range, Value value)
     }
 }
 
+import std.range : ElementEncodingType;
+import std.traits : isSomeChar, isMutable;
+
+/// ditto
+void fill(A, Value)(auto ref A array, auto ref Value value)
+if (isArray!A && isSomeChar!(ElementEncodingType!A) &&
```

Comment:
> . I'm tempted to use a is(typeof((){array[] = value})) to avoid a heavy constraint..
> Would you have a better constraint as example ?

I think the constraint is quite rare, so that's also the best way that comes to my mind and it's still quite readable as API documentation  ;-)